stock_inventory/models/stock_inventory.py

```diff
@@ -191,25 +191,45 @@ def _get_domain_category_quants(self, base_domain):
             ]
         )
 
+    def _get_products(self):
+        self.ensure_one()
+        if self.product_selection == "all":
+            return self.env["product.product"].search([])
+        elif self.product_selection == "category" and self.category_id:
+            return self.env["product.product"].search(
+                [("categ_id", "=", self.category_id.id)]
+            )
```

Comment:
@AaronHForgeFlow one last comment: should we consider also child categories here?

Reply:
good point!